lib/tss/response_handlers.rb

```diff
@@ -4,11 +4,22 @@ module ResponsesHandlers
 
   def handle_response(verb, response, path, root_name = nil)
     case response.code
-    when 200..299
-      response_json = JSON.parse(response.body)
-      root_name.nil? ? response_json : response_json[root_name]
+    when 200..299 then handle_response_success(response, root_name)
     when 404 then nil
+    when 422 then handle_response_unprocessable_entity
     else raise "TSS: '#{verb} #{path}' failed with status = #{response.code}."
     end
   end
+
+  def handle_response_success(response, root_name = nil)
+    response_json = JSON.parse(response.body)
+    root_name.nil? ? response_json : response_json[root_name]
+  end
+
+  def handle_response_unprocessable_entity(verb, response, path)
+    response_json = JSON.parse(response.body)
+    response_json = response_json['error'] if response_json.key?('error')
+    raise "TSS: '#{verb} #{path}' failed with status = #{response.code}, \
+message: #{response_json}."
```

Comment:
Currently, on the core project, if there was an error while saving (any kind of error) I render the edit form, and the error message (422, 500). In case of error 500 it could be anything, like timeout error, so maybe we should handle this similar to the 422.

Or I can create a new error class in the payment-connector, and use this in case of unprocessable entity.